src/components/molecules/header.rs

```diff
@@ -116,6 +116,18 @@ pub fn Header() -> Element {
 
     let on_handle_account = use_coroutine(move |mut rx: UnboundedReceiver<u8>| async move {
         while let Some(event) = rx.next().await {
+            let accounts_list = accounts.get();
+
+            if accounts_list.is_empty() {
+                log::warn!("Accounts list is empty");
+                return notification.handle_warning(&translate!(i18, "warnings.middlewares.not_account"));
+            }
+
+            if (event as usize) >= accounts_list.len() {
+                log::warn!("The index is out of bounds");
+                return;
+            }
+
             let account = &accounts.get()[event as usize];
```

Comment:
If I understand correctly, the code was panicking because for some reason `event` was an index not existing in the accounts vec? in that case I would suggest just ["safely accessing"](https://doc.rust-lang.org/std/primitive.slice.html#method.get) the elements of the vec.
```rust
let Some(account) = account_list.get(event as usize) else {
    return notification.handle_warning(&translate!(i18, "warnings.middlewares.not_account"));
};
```